xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -133,15 +136,15 @@ impl EncryptedMessageStore {
     fn raw_conn(
         &self,
     ) -> Result<PooledConnection<ConnectionManager<SqliteConnection>>, StorageError> {
-        let mut conn = self
+        let pool = self
             .pool
-            .get()
+            .read()
             .map_err(|e| StorageError::Pool(e.to_string()))?;
```

Comment:
It would be better practice to create a new error variant specific to locking, just in case a codepath upstream wants to handle poisons, as well as makes debugging more precise. Maybe like a Lock(#[from] PoisonError) variant on StorageError makes sense